framework/src/modules/chain/chain.js

```diff
@@ -272,6 +295,40 @@ module.exports = class Chain {
 					action.params[0],
 					action.params[1]
 				),
+			blocks: async action =>
+				new Promise(resolve => {
+					// Get 34 blocks with all data (joins) from provided block id
+					// According to maxium payload of 58150 bytes per block with every transaction being a vote
+					// Discounting maxium compression setting used in middleware
+					// Maximum transport payload = 2000000 bytes
+					const query = action.params[0] || {};
+					this.scope.modules.blocks.utils.loadBlocksData(
+						{
+							limit: 34, // 1977100 bytes
```

Comment:
We should research and update this number. If I remember well it's not accurate. Block size varies depending on the transaction associated to it. We might be able to increase the number significantly.

Reply:
We should raise a new issue.

Reply:
For this current protocol, let's keep it this way because it's not really a protocol change, and we don't want to risk here.
However, please create an issue to address this, and we can tackle this when we change the p2p protocol

Reply:
@jondubois would you mind to open the issue?